app/controllers/discourse_rss_polling/feed_settings_controller.rb

```diff
@@ -10,21 +10,46 @@ def show
 
     def update
       if params[:feed_settings] == []
-        new_feed_settings = []
+        RssFeed.destroy_all
       else
-        new_feed_settings =
-          (feed_setting_params.presence || []).map do |feed_setting|
-            feed_setting.values_at(
-              :feed_url,
-              :author_username,
-              :discourse_category_id,
-              :discourse_tags,
-              :feed_category_filter,
-            )
+        # Temporary until we start using IDs from the db
+        # and can update individual items
+        if feed_setting_params.presence
+          current_feeds = RssFeed.all.to_a
+          feed_setting_params.each do |feed|
+            current_feeds.delete_if do |h|
+              h.url == feed["feed_url"] && h.category_id == feed["discourse_category_id"].to_i &&
+                h.category_filter == feed["feed_category_filter"]
+            end
+            rss_feed =
+              RssFeed.find_by(
+                url: feed["feed_url"],
+                category_id: feed["discourse_category_id"],
+                category_filter: feed["feed_category_filter"],
+              )
+            if rss_feed
+              rss_feed.update!(
+                url: feed["feed_url"],
+                author: feed["author_username"],
+                category_id: feed["discourse_category_id"],
+                tags: feed["discourse_tags"]&.join(","),
+                category_filter: feed["feed_category_filter"],
+              )
+            else
+              RssFeed.create!(
+                url: feed["feed_url"],
+                author: feed["author_username"],
+                category_id: feed["discourse_category_id"],
+                tags: feed["discourse_tags"]&.join(","),
+                category_filter: feed["feed_category_filter"],
+              )
+            end
           end
-      end
 
-      SiteSetting.rss_polling_feed_setting = new_feed_settings.to_yaml
+          # Delete any remaining feeds
+          current_feeds&.each { |feed| RssFeed.destroy_by(id: feed.id) }
```

Comment:
I think this can be moved into the `if` block above? (and then `&.` can be changed to `.`)